### PR TITLE
API: Enable gzip compression

### PR DESCRIPTION
**Pull request by gabrielmougard**
And add security response headers to documentation endpoint.

Fixes https://github.com/canonical/lxd/issues/13150

### CODE REVIEW

File: lxd/documentation.go [hunk 1]
```diff
@@ -58,11 +58,10 @@ func metadataConfigurationGet(d *Daemon, r *http.Request) response.Response {
...
 
-	var data map[string]any
-	err = json.Unmarshal(file, &data)
+	formattedJSON, err := json.MarshalIndent(json.RawMessage(file), "", "  ")
 	if err != nil {
 		return response.SmartError(err)
 	}
 
-	return response.SyncResponse(true, data)
+	return response.SyncResponsePlain(true, true, string(formattedJSON))
```
Comment (tomponline) [inline lxd/documentation.go:66]: this changes the response output format doesn't it?

Comment (gabrielmougard) [reply lxd/documentation.go:66]: So here how it was before:
before-metadata


And here how it is with the change:
after-metadata

Comment (gabrielmougard) [reply lxd/documentation.go:66]: I don't exactly know how this API endpoint is consumed but I'd assume that a Javascript client has no problem dealing with that. Correct me @edlerd if I'm wrong. But this way is more human readable is gzip encoded.

Comment (edlerd) [reply lxd/documentation.go:66]: Yes, I am confident the JSON parser will ignore the extra whitespace.

Personally, I am fine with or without the extra formatting. I always use the network tab of the browser to explore JSON responses. There the structure is parsed and presented in a nice format.

Comment (markylaing) [reply lxd/documentation.go:66]: I don't think `SyncResponsePlain` adds the `content-type` header with `appliation/json` which this endpoint probably should have.

Comment (gabrielmougard) [reply lxd/documentation.go]: fixed

**Review by tomponline — APPROVED** [review]

**Comment by tomponline** [comment]
@edlerd this should be in latest/edge now please can you check the UI is working OK with it?

Thanks

**Pull request merged**